packages/hub/tasks/send-notifications.ts

```diff
@@ -1,7 +1,90 @@
+import { inject } from '@cardstack/di';
 import { Helpers } from 'graphile-worker';
+import * as Sentry from '@sentry/node';
+
+export interface PushNotificationData {
+  /**
+   * A unique ID determined by the caller of this task, to be used for deduplication
+   */
+  notificationId: string;
+  notificationType: string;
+  pushClientId: string;
+  notificationTitle?: string;
+  notificationBody?: string;
+  notificationData?: {};
+}
+
+export interface PushNotificationsIdentifiers {
+  notificationId: PushNotificationData['notificationId'];
+}
 
 export default class SendNotificationsTask {
-  async perform(payload: any, helpers: Helpers) {
-    helpers.logger.info('Notification to send:', payload);
+  sentPushNotificationsQueries = inject('sent-push-notifications-queries', { as: 'sentPushNotificationsQueries' });
+  firebasePushNotifications = inject('firebase-push-notifications', { as: 'firebasePushNotifications' });
+
+  async perform(payload: PushNotificationData, helpers: Helpers) {
+    try {
+      // we do this in a separate block because failure to read should not affect
```

Comment:
I think this is a bit too defensive - a failure to interact with the database would imply a much deeper problem and I think it shouldn't be a concern of the worker to handle that. Ignoring that the database is failing and still performing work can lead to undesirable states which are hard to debug. 

I think it would be better if we just let the worker fail and let it be retried.

Reply:
Currently not separating the try catch block since db services + the worker are both reliant on the db.. and the hub's subscriptions also rely on writing to the db succeeding. 

Maybe we can re-examine this at some point, in https://linear.app/cardstack/issue/CS-2806/review-push-notifications-dependency-on-database-services?